docs/tls/README.md

```diff
@@ -1,90 +1,145 @@
+This section describes how to use a private key and public certificate that have been obtained from a certificate authority (CA). If these files have not been obtained, skip to [3. Generate Self-signed Certificates](#genselfsignedcerts) or generate them with [Let's Encrypt](https://letsencrypt.org) using these instructions: [https://docs.minio.io/docs/generate-let-s-encypt-certificate-using-concert-for-minio](https://docs.minio.io/docs/).
 
-If you're looking to generate CA certificate for Minio using Let's Encrypt, follow the docs [here](https://docs.minio.io/docs/generate-let-s-encypt-certificate-using-concert-for-minio).
+Copy the existing private key and public certificate to the `certs` directory within the Minio configuration directory. The default configuration directory is:
+* **Linux:** `${HOME}/.minio/`
+* **Windows:** `%%USERPROFILE%%\.minio\`
 
-## 3. Generate self-signed certificates
+**Note:** 
+* The key and certificate files must be appended with `.key` and `.crt`, respectively.
+* A filename for a certificate signed by a CA consists of a concatenation of the server's certificate, any intermediates, and the CA's root certificate.
 
-Before generating your self-signed certificate, note that
+## <a name="genselfsignedcerts"></a>3. Generate and use Self-signed Keys and Certificates with Minio
 
-- Minio supports only key/certificate in PEM format on Linux.
-- Minio supports only key/certificate in PEM format on Windows. We don't support PFX certificates currently.
+This section describes how to generate a self-signed certificate using various tools:
 
-### Using generate_cert.go
+3.1 [Use generate_cert.go to Generate a Certificate](#usinggo) 
+3.2 [Use OpenSSL to Generate a Certificate](#usingopenssl) 
+3.3 [Use OpenSSL (with IP address) to Generate a Certificate](#usingopensslwithip) 
+3.4 [Use GnuTLS (for Windows) to Generate a Certificate](#usinggnutls)
 
-Download [generate_cert.go](https://golang.org/src/crypto/tls/generate_cert.go?m=text). This is a simple go tool to generate self-signed certificates.
+**Note:**
+* Minio only supports keys and certificates in PEM format on Linux and Windows.
+* Minio doesn't currently support PFX certificates.
 
-`generate_cert.go` already provides SAN certificates with DNS and IP entries:
+### <a name="usinggo"></a>3.1 Use generate_cert.go to Generate a Certificate
+
+Download [`generate_cert.go`](https://golang.org/src/crypto/tls/generate_cert.go?m=text).
+
+`generate_cert.go` is a simple *Go* tool to generate self-signed certificates, and provides SAN certificates with DNS and IP entries:
 
 ```sh
 go run generate_cert.go -ca --host "10.10.0.3"
 ```
 
-### Using OpenSSL
+A response similar to this one should be displayed:
+
+```
+2018/11/21 10:16:18 wrote cert.pem
+2018/11/21 10:16:18 wrote key.pem
+```
+
+### <a name="usingopenssl"></a>3.2 Use OpenSSL to Generate a Certificate
```

Comment:
this would be use-openssl-to-generate-certificate. Doing this makes it long but becomes friendly URLs when sharing.